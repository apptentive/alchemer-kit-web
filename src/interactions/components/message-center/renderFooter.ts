/**
 * Renders a given Message Center footer.
 *
 * @returns {HTMLElement} - The footer DOM Node.
 */
const renderFooter = () => {
  const footer = document.createElement('footer');
  footer.classList.add('apptentive-footer');

  return footer;
};

export default renderFooter;
