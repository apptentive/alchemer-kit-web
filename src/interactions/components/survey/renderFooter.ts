interface IFooterOptions {
  contactUrl?: string;
  contactUrlText?: string;
  termsAndConditionsUrl?: string;
  termsAndConditionsText?: string;
}

/**
 * Renders a given surveys footer.
 *
 * @param {object} configuration - The configuration object to drive the footer creation
 * @returns {HTMLElement} - The DOM Node
 */
const renderFooter = (configuration: IFooterOptions) => {
  const footer = document.createElement('footer');
  footer.className = 'apptentive-survey-footer';

  if (configuration.contactUrl && configuration.contactUrlText) {
    const contact = document.createElement('a');
    contact.className = 'apptentive-survey-footer__link';
    contact.textContent = configuration.contactUrlText;
    contact.href = configuration.contactUrl;
    contact.target = '_blank';

    footer.append(contact);
  }

  if (configuration.termsAndConditionsUrl && configuration.termsAndConditionsText) {
    const termsAndConditions = document.createElement('a');
    termsAndConditions.className = 'apptentive-survey-footer__link';
    termsAndConditions.textContent = configuration.termsAndConditionsText;
    termsAndConditions.href = configuration.termsAndConditionsUrl;
    termsAndConditions.target = '_blank';

    footer.append(termsAndConditions);
  }

  return footer;
};

export default renderFooter;
