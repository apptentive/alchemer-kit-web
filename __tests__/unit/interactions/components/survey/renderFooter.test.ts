import renderFooter from '../../../../../src/interactions/components/survey/renderFooter';

const _contactUrl = 'https://test.com';
const _contactUrlText = 'Contact Us';
const _termsAndConditionsUrl = 'https://tandc.com';
const _termsAndConditionsText = 'Terms & Conditions';

describe('renderFooter', () => {
  test('properly renders footer', () => {
    const output = renderFooter({
      contactUrl: _contactUrl,
      contactUrlText: _contactUrlText,
      termsAndConditionsUrl: _termsAndConditionsUrl,
      termsAndConditionsText: _termsAndConditionsText,
    });

    expect(output).toMatchSnapshot();
  });

  test('properly renders footer without terms and conditions config', () => {
    const output = renderFooter({
      contactUrl: _contactUrl,
      contactUrlText: _contactUrlText,
    });

    expect(output).toMatchSnapshot();
  });

  test('properly renders footer with partial terms and conditions config', () => {
    const output = renderFooter({
      contactUrl: _contactUrl,
      contactUrlText: _contactUrlText,
      termsAndConditionsUrl: _termsAndConditionsUrl,
    });

    expect(output).toMatchSnapshot();
  });

  test('properly renders footer without contact config', () => {
    const output = renderFooter({
      termsAndConditionsUrl: _termsAndConditionsUrl,
      termsAndConditionsText: _termsAndConditionsText,
    });

    expect(output).toMatchSnapshot();
  });

  test('properly renders footer with partial contact config', () => {
    const output = renderFooter({
      contactUrl: _contactUrl,
      termsAndConditionsUrl: _termsAndConditionsUrl,
      termsAndConditionsText: _termsAndConditionsText,
    });

    expect(output).toMatchSnapshot();
  });
});
