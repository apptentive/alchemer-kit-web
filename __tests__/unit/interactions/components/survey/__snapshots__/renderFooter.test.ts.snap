// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderFooter properly renders footer 1`] = `
<footer
  class="apptentive-survey-footer"
>
  <a
    class="apptentive-survey-footer__link"
    href="https://test.com"
    target="_blank"
  >
    Contact Us
  </a>
  <a
    class="apptentive-survey-footer__link"
    href="https://tandc.com"
    target="_blank"
  >
    Terms & Conditions
  </a>
</footer>
`;

exports[`renderFooter properly renders footer with partial contact config 1`] = `
<footer
  class="apptentive-survey-footer"
>
  <a
    class="apptentive-survey-footer__link"
    href="https://tandc.com"
    target="_blank"
  >
    Terms & Conditions
  </a>
</footer>
`;

exports[`renderFooter properly renders footer with partial terms and conditions config 1`] = `
<footer
  class="apptentive-survey-footer"
>
  <a
    class="apptentive-survey-footer__link"
    href="https://test.com"
    target="_blank"
  >
    Contact Us
  </a>
</footer>
`;

exports[`renderFooter properly renders footer without contact config 1`] = `
<footer
  class="apptentive-survey-footer"
>
  <a
    class="apptentive-survey-footer__link"
    href="https://tandc.com"
    target="_blank"
  >
    Terms & Conditions
  </a>
</footer>
`;

exports[`renderFooter properly renders footer without terms and conditions config 1`] = `
<footer
  class="apptentive-survey-footer"
>
  <a
    class="apptentive-survey-footer__link"
    href="https://test.com"
    target="_blank"
  >
    Contact Us
  </a>
</footer>
`;
